ocaml/xapi/xapi_network_sriov_helpers.ml

```diff
@@ -152,11 +152,7 @@ let get_sriov_networks_from_vm ~__context ~vm =
 let get_local_underlying_pif ~__context ~network ~host =
   match Xapi_network_attach_helpers.get_local_pifs ~__context ~network ~host with
     | [] -> raise Api_errors.(Server_error (internal_error, ["Cannot get local pif on network"]))
```

Comment:
I think you should make this return None as well rather than an exception. There are networks without PIFs as well (internal networks).

Reply:
Hi, @robhoes  I had updated. I think here we indeed should return `None` for internal networks without PIFs as you mentioned. But also if the network is not _empty_, we should raise an exception because this means the host doesn't have the ability for running the VM with that `network`.
Assume VM with VGPU and SR-IOV VIF, the host was chosen by VGPU resources then we need to assert the VM can run on this host, if the selected host do not have any PIF on that `network`,  just return `None` for _get_local_pifs_ will cause the assert makes no sense.
Please take a look at the latest changes. Thanks.

Reply:
There is some overlap here with `assert_can_see_networks`, which is called in `assert_can_boot_here` before `assert_netsriov_available`. This function already checks whether the host has the PIFs that are needed by the VIFs of the VM. It seems to me that `assert_netsriov_available` should focus on just checking the available _capacity_ on the SR-IOV PIFs.

Reply:
Yes, you are right. I didn't noticed `assert_can_see_networks` before, thanks. 
I will make `assert_netsriov_available` clean to just return an option without any internal exception.
Thanks, Rob!

Reply:
Updated, please take a look again. Thanks a lot. @robhoes